test/Broadway/EventHandling/SimpleEventBusTest.php

```diff
@@ -104,6 +104,41 @@ public function it_does_not_dispatch_new_events_before_all_listeners_have_run()
         $this->eventBus->publish($domainEventStream);
     }
 
+    /**
+     * @test
+     */
+    public function it_should_still_publish_events_after_exception()
+    {
+        $domainMessage1 = $this->createDomainMessage(array('foo' => 'bar'));
+        $domainMessage2 = $this->createDomainMessage(array('foo' => 'bas'));
+
+        $domainEventStream1 = new DomainEventStream(array($domainMessage1));
+        $domainEventStream2 = new DomainEventStream(array($domainMessage2));
+
+        $eventListener = $this->createEventListenerMock();
+        $eventListener
+            ->expects($this->at(0))
```

Comment:
Ah I see. I vaguely recall everzet saying something similar in a talk advocating for the use of Prophecy in place of PHPUnit mocks, but since I never experienced this pain first hand it didn't really stick.

I agree that the interface of listeners mitigate this greatly in this case. But thanks for the insight :).